deploy/helm/nfd-operator/templates/_helpers.tpl

```diff
@@ -0,0 +1,74 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "node-feature-discovery-operator.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Allow the release namespace to be overridden for multi-namespace deployments in combined charts
+*/}}
+{{- define "node-feature-discovery-operator.namespace" -}}
+  {{- if .Values.namespaceOverride -}}
+    {{- .Values.namespaceOverride -}}
+  {{- else -}}
+    {{- .Release.Namespace -}}
+  {{- end -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "node-feature-discovery-operator.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
```

Comment:
Just a stupid question to everyone. A one that now caught my eye – and I know we have exactly similar thing in NFD and I've lived with it. But, why do we have 

```
{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
```

instead of 

```
{{- printf "%s-%s" $name .Release.Name | trunc 63 | trimSuffix "-" }}
```

Which would somehow feel more logical to me. @adrianchiris any comment on this?

Somehow `nfd-operator-foobarxyz` would be more understandable and readable to me that `foobarxyz-nfd-operator`.

Reply:
helm can bootstrap a basic chart using `helm create <chart-name`
these helper templates are automatically generated.

afterwhich you can edit add etc...

the fullname here is just the default impl created by helm create command.
it can be changed if desired. and is eventually a matter of taste IMO.

Reply:
Ah ok, that explains why it's like that everywhere. Thanks for educating me 🤓 Case closed